Include/cpython/listobject.h

```diff
@@ -40,6 +40,12 @@ static inline Py_ssize_t PyList_GET_SIZE(PyObject *op) {
 
 static inline void
 PyList_SET_ITEM(PyObject *op, Py_ssize_t index, PyObject *value) {
+    if(_Py_IsImmutable(op)){
```

Comment:
I can't, because it is not currently part of the public API. I may want to move it there, but ultimately we'll replace this code entirely as the Python folks won't want a test in this codepath as it "should only be called by consumers of the C ABI during list init" and it is the only place that needs the check that isn't internal. The real SetItem() has an internal implementation that calls CHECKWRITE approlpriately.